src/main/java/creativei/service/impl/FollowUpServiceImpl.java

```diff
@@ -7,23 +7,58 @@
 import creativei.exception.DataIntegrityException;
 import creativei.exception.InvalidParamRequest;
 import creativei.service.FollowUpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Arrays;
+
 @Service("FollowUpService")
 public class FollowUpServiceImpl implements FollowUpService {
+    private static final Logger logger= LoggerFactory.getLogger(FollowUpService.class);
     @Autowired
     FollowUpDao followUpDao;
 
     @Override
     public FollowUp createFollowUp(FollowUp followUp) throws InvalidParamRequest {
         if (!validateFollowUp(followUp))
-            throw new InvalidParamRequest("Required field can not be empty");
+            throw new InvalidParamRequest("Required field are either null or does not contain the appropirate value");
+        return followUpDao.save(followUp);
+    }
 
+    @Override
+    public FollowUp updateFollowUp(FollowUp followUp) throws InvalidParamRequest {
+        if (!validateFollowUp(followUp))
+            throw new InvalidParamRequest("Required field are either null or does not contain the appropirate value");
         return followUpDao.save(followUp);
     }
+
     private Boolean validateFollowUp(FollowUp followUp) {
-        if (followUp.getCaseIndex() == null || followUp.getFollowUpDate() == null || followUp.getType() == null || followUp.getStatus() == null || followUp.getSubStatus() == null) {
+        Boolean []cond=new Boolean[5];
+        Arrays.fill(cond,false);
+        if (((followUp.getCaseIndex() == null) && (cond[0]=true)) | (followUp.getFollowUpDate() == null&& (cond[1]=true)) | (followUp.getType() == null&& (cond[2]=true)) | (followUp.getStatus() == null && (cond[3]=true)) | (followUp.getSubStatus() == null&& (cond[4]=true))) {
+            for(int i=0;i<cond.length;i++){
```

Comment:
This is not a good algorithm. Avoid double checking. Use separate conditions for each validation and avoid loop and swith case

Reply:
issue resolved in 21a3eac